### PR TITLE
Remove delegation and implicit accounts amounts from available balance

### LINKED ISSUE #2170: Update available balance for base coin
## Description

Now the amount of `implicit account outputs` and `delegation outputs` is used to calculate the `available balance` of the `base coin`

## Motivation

Return an `available balance` more accurate, without using the amount of `implicit account outputs` and `delegation outputs`

## Requirements

Write a list of what you want this feature to do.

1. Return an available balance without using `implicit accounts` and `delegations` outputs of the wallet

## Are you planning to do it yourself in a pull request?

No

### PR DESCRIPTION
**Pull request by Thoralf-M**
# Description of change

Remove delegation and implicit accounts amounts from available balance

## Links to any relevant issues

Fixes https://github.com/iotaledger/iota-sdk/issues/2170

## Type of change

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

With the cli wallet, balance before output with implicit account creation address:
```
    base_coin: BaseCoinBalance {
        total: 2000000000,
        available: 1999968300,
```
After an output with implicit account creation address got added
```
    base_coin: BaseCoinBalance {
        total: 3000000000,
        available: 1999968300,
```

### CODE REVIEW

File: sdk/src/wallet/operations/balance.rs [hunk 1]
```diff
@@ -338,13 +346,18 @@ impl<S: 'static + SecretManage> Wallet<S> {
 
         #[cfg(not(feature = "participation"))]
         {
-            balance.base_coin.available = balance.base_coin.total.saturating_sub(locked_amount);
+            balance.base_coin.available = balance
+                .base_coin
+                .total
+                .saturating_sub(delegation_implicit_accounts_amount)
```
Comment (DaughterOfMars) [inline sdk/src/wallet/operations/balance.rs]: Why not just not add it above in the first place?

Comment (Thoralf-M) [reply sdk/src/wallet/operations/balance.rs]: Because it's still part of the total balance

Comment (Thoralf-M) [reply sdk/src/wallet/operations/balance.rs]: Or do you mean something else?

Comment (DaughterOfMars) [reply sdk/src/wallet/operations/balance.rs]: Ah, I see

**Review by DaughterOfMars — DISMISSED** [review]

File: sdk/src/wallet/operations/balance.rs [hunk 2]
```diff
@@ -338,13 +346,18 @@ impl<S: 'static + SecretManage> Wallet<S> {
...
+                .saturating_sub(locked_amount);
```
Comment (thibault-martinez) [inline sdk/src/wallet/operations/balance.rs]: Would it not make sense to consider it locked?

Comment (thibault-martinez) [reply sdk/src/wallet/operations/balance.rs]: I mean, adding amounts to locked_amount directly

Comment (Thoralf-M) [reply sdk/src/wallet/operations/balance.rs]: Sure, done

File: sdk/src/wallet/operations/balance.rs [hunk 3]
```diff
@@ -133,6 +138,9 @@ impl<S: 'static + SecretManage> Wallet<S> {
 
                         // Add amount
                         balance.base_coin.total += output.amount();
+                        if address_unlock_cond.address().kind() == ImplicitAccountCreationAddress::KIND {
```
Comment (thibault-martinez) [inline sdk/src/wallet/operations/balance.rs]: Use `is_implicit_account_creation` instead

Comment (Thoralf-M) [reply sdk/src/wallet/operations/balance.rs]: Ah nice didn't see this and thought we only have the on that returns an Option

**Review by thibault-martinez — APPROVED** [review]

**Review by DaughterOfMars — APPROVED** [review]

**Pull request merged**